mods/legacy-shell-message-boxes.wh.cpp

```diff
@@ -0,0 +1,256 @@
+        LocalFree(pszMsg);
+    }
+    else
+    {
+        g_dwSMBThreadID = (DWORD)-1;
+        SetLastError(ERROR_OUTOFMEMORY);
+    }
+    va_end(args);
+    return nRet;
+}
+
+using ShellMessageBoxA_t = decltype(&ShellMessageBoxA);
+ShellMessageBoxA_t ShellMessageBoxA_orig;
+int WINAPI ShellMessageBoxA_hook(
+    HINSTANCE hAppInst,
+    HWND      hWnd,
+    LPCSTR    lpcText,
+    LPCSTR    lpcTitle,
+    UINT      fuStyle,
+    ...
+)
+{
+    g_dwSMBThreadID = GetCurrentThreadId();
+    va_list args;
+    va_start(args, fuStyle);
+    LPSTR pszMsg = ConstructMessageStringA(hAppInst, lpcText, &args);
+    int nRet = 0;
+    if (pszMsg)
+    {
+        nRet = ShellMessageBoxA_orig(hAppInst, hWnd, pszMsg, lpcTitle, fuStyle);
+        LocalFree(pszMsg);
+    }
+    else
+    {
+        g_dwSMBThreadID = (DWORD)-1;
+        SetLastError(ERROR_OUTOFMEMORY);
+    }
+    va_end(args);
+    return nRet;
+}
+
+const WindhawkUtils::SYMBOL_HOOK shlwapiDllHooks[] = {
+    {
+        {
+#ifdef _WIN64
+            L"ConstructMessageStringW"
+#else
+            L"_ConstructMessageStringW@12"
+#endif
+        },
+        &ConstructMessageStringW,
+        nullptr,
+        false
+    },
+    {
+        {
+#ifdef _WIN64
+            L"ConstructMessageStringA"
+#else
+            L"_ConstructMessageStringA@12"
+#endif
+        },
+        &ConstructMessageStringA,
+        nullptr,
+        false
+    }
+};
+
+VS_FIXEDFILEINFO* GetModuleVersionInfo(HMODULE hModule, UINT *puPtrLen) 
+{ 
+    void *pFixedFileInfo = nullptr; 
+    UINT uPtrLen = 0; 
+
+    HRSRC hResource = 
+        FindResourceW(hModule, MAKEINTRESOURCEW(VS_VERSION_INFO), RT_VERSION); 
+    if (hResource)
+    { 
+        HGLOBAL hGlobal = LoadResource(hModule, hResource); 
+        if (hGlobal)
+        { 
+            void *pData = LockResource(hGlobal); 
+            if (pData)
+            { 
+                if (!VerQueryValueW(pData, L"\\", &pFixedFileInfo, &uPtrLen)
+                || uPtrLen == 0)
+                { 
+                    pFixedFileInfo = nullptr; 
+                    uPtrLen = 0; 
+                } 
+            } 
+        } 
+    } 
+
+    if (puPtrLen)
+    { 
+        *puPtrLen = uPtrLen; 
+    } 
+  
+     return (VS_FIXEDFILEINFO *)pFixedFileInfo; 
+ } 
+
+/**
+  * Loads comctl32.dll, version 6.0.
+  * This uses an activation context that uses shell32.dll's manifest
+  * to load 6.0, even in apps which don't have the proper manifest for
+  * it.
+  */
+HMODULE LoadComCtlModule(void)
+{
+    HMODULE hShell32 = LoadLibraryW(L"shell32.dll");
+    ACTCTXW actCtx = { sizeof(actCtx) };
+    actCtx.dwFlags = ACTCTX_FLAG_RESOURCE_NAME_VALID | ACTCTX_FLAG_HMODULE_VALID;
+    actCtx.lpResourceName = MAKEINTRESOURCEW(124);
+    actCtx.hModule = hShell32;
+    HANDLE hActCtx = CreateActCtxW(&actCtx);
+    ULONG_PTR ulCookie;
+    ActivateActCtx(hActCtx, &ulCookie);
+    HMODULE hComCtl = LoadLibraryW(L"comctl32.dll");
```

Comment:
Can you please use `LoadLibrarExW` as discussed recently in https://github.com/ramensoftware/windhawk-mods/issues/2063?